### DISCUSSION TITLE
On demand feature in pg_timetable

### DISCUSSION THREAD

**Discussion by ashx1901** [cybertec-postgresql/pg_timetable#478] [Q&A] [answered]
Hi,

I am new to pg_timetable and we are in a process of migrating our databases into Postgres. I am looking for a feature in pg_timetable where the job can be run on-demand. Can anyone please help me to get this done?

Thanks

**Comment by pashagolub** [comment]
Hello, and thank you for your interest in pg_timetable.

Under "on-demand" you mean the operator can start any chain at any time, right?

**Reply by ashx1901** [reply]
Yes, If i want to start any job out of its schedule how can achieve that? 

Thanks

**Comment by pashagolub** [comment, marked as answer]
One can start any chain on demand and stop any running chain. There are two functions available:
* `timetable.notify_chain_start(
    chain_id BIGINT, 
    worker_name TEXT
)`
* `timetable.notify_chain_stop(
    chain_id BIGINT, 
    worker_name TEXT
)`

So the algorithm is:
1. Make sure `pg_timetable` is running with a proper client name, e.g. `--clientname=worker01`
2. Connect to the target database with the preferred client, e.g. `psql -d timetable -h localhost -U scheduler`
3. Inspect what chain needs to be run, e.g., `SELECT * FROM timetable.chain`
4. Execute the function to send the chain to the worker, e.g., `SELECT timetable.notify_chain_start(42, 'worker01')`
5. Check the logs for the chain output

**Reply by ashx1901** [reply]
Thank you so much for the above solution.
